pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -1159,3 +1169,9 @@ func toSnakeCase(s string) string {
 	snake := matchAllCap.ReplaceAllString(s, "${1}_${2}")
 	return strings.ToLower(snake)
 }
+
+func toSingularResourceName(s string) string {
```

Comment:
This behavior will likely be changed in the future (i.e. relying on https://github.com/DataDog/datadog-agent/pull/33213).